packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -166,6 +176,61 @@ export class GutenbergEditorPage extends BaseContainer {
 		return lines.filter( Boolean ).join( '\n' );
 	}
 
+	/**
+	 * Given a name, adds the Gutenberg block matching the name.
+	 *
+	 * The name is expected to be formatted in the same manner as it
+	 * appears on the label when visible in the block inserter panel.
+	 *
+	 * Example:
+	 * 		- Click to Tweet
+	 * 		- Pay with Paypal
+	 * 		- SyntaxHighlighter Code
+	 *
+	 * @param {string} blockName Name of the block to be inserted.
+	 */
+	async addBlock( blockName: string ): Promise< ElementHandle > {
+		const isBlockInserterOpen = await this.frame.$eval(
+			selectors.blockInserterToggle,
+			( element ) => element.classList.contains( 'is-pressed' )
+		);
+		// Open the block inserter panel if not open already.
+		if ( ! isBlockInserterOpen ) {
+			await this.openBlockInserter();
+		}
+
+		await this.frame.fill( selectors.blockSearch, blockName );
+
+		await this.frame.click( `${ selectors.blockInserterResultItem }:has-text("${ blockName }")` );
+		// Confirm the block has been added to the editor body.
+		return await this.frame.waitForSelector( `[aria-label="Block: ${ blockName }"]` );
+	}
+
+	/**
+	 * Open the block inserter panel.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async openBlockInserter(): Promise< void > {
+		await this.frame.click( selectors.blockInserterToggle );
+		await this.frame.waitForSelector( selectors.blockInserterPanel );
+	}
+
+	/**
+	 * Opens the settings sidebar.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async openSidebar(): Promise< void > {
```

Comment:
The name `sidebar` may not be the best here, but I could not glean any useful information from the CSS selector. Any thoughts?

Reply:
Sounds good.